src/env.rs

```diff
@@ -602,3 +617,61 @@ impl Environment {
         &self.profile_target_dir
     }
 }
+
```

Comment:
:thought_balloon: I don't think this needs to be done, but it's something to consider.

Instead of using mock one could create a test method like below that provides a `temp_env`
This would limit the sprinkling of `cfg(test)` to only this function call, and iff the `cargo_build.rs` test needs to get more complex it can make it's own test_env method.

```rust
#[cfg(test)]
pub fn test_env() -> Environment {
    let expected_out_dir = "/x86_64-unknown-linux-gnu/path_to_out_directory";
    let expected_target = "x86_64-unknown-linux-gnu";
    let expected_cargo_path = "/path_to_cargo";
    let expected_profile = "debug";
    let expected_cargo_package_version = "2.1.0-pre0";
    temp_env::with_vars(
        [
            ("OUT_DIR", Some(expected_out_dir)),
            ("TARGET", Some(expected_target)),
            ("PROFILE", Some(expected_profile)),
            ("CARGO", Some(expected_cargo_path)),
            ("HOST", Some("host")),
            ("NUM_JOBS", Some("11")),
            ("OPT_LEVEL", Some("2")),
            ("RUSTC", Some("rustc")),
            ("RUSTDOC", Some("rustdoc")),
            ("CARGO_PKG_VERSION", Some(expected_cargo_package_version)),
            ("CARGO_PKG_AUTHORS", Some("MobileCoin")),
            ("CARGO_PKG_NAME", Some("mc-build-rs")),
            ("CARGO_PKG_DESCRIPTION", Some("")),
            ("CARGO_PKG_HOMEPAGE", Some("")),
            ("CARGO_PKG_REPOSITORY", Some("")),
            ("CARGO_CFG_TARGET_ARCH", Some("x86_64")),
            ("CARGO_CFG_TARGET_ENDIAN", Some("little")),
            ("CARGO_CFG_TARGET_ENV", Some("")),
            ("CARGO_CFG_TARGET_FAMILY", Some("unix")),
            ("CARGO_CFG_TARGET_FEATURE", Some("adx,aes,avx,avx2,")),
            ("CARGO_CFG_TARGET_OS", Some("linux")),
            ("CARGO_CFG_TARGET_POINTER_WIDTH", Some("64")),
            ("CARGO_CFG_TARGET_VENDOR", Some("unknown")),
        ],
        || {
            Environment::default() } )
}
```
in `cargo_build.rs`, if one wants the function could be expanded like the `init_mock()`
```rust
        let env = test_env();

        let cargo_builder = CargoBuilder::new(&env, &Path::new(expected_output_dir), false);
```